client/blocks/reader-avatar/index.tsx

```diff
@@ -113,22 +113,25 @@ export default function ReaderAvatar( {
 		'has-site-icon': hasSiteIcon,
 		'has-gravatar': hasAvatar || showPlaceholder,
 	} );
-	const defaultIconElement = ! hasSiteIcon && ! hasAvatar && ! showPlaceholder && (
```

Comment:
(Not related to warnings) Moved all the `siteIcon` related functionality in this code block.